python/pyspark/pandas/typedef/typehints.py

```diff
@@ -559,6 +559,9 @@ def infer_return_type(f: Callable) -> Union[SeriesType, DataFrameType, ScalarTyp
 
     tpe = get_type_hints(f).get("return", None)
 
+    if tpe is None:
+        raise TypeError("A return value is required for the input function")
```

Comment:
Technically, I think it should be ValueError, because func is `Callable` but the value is not not meet the requirement. See also in why raise TypeError before (it just because some compatiable reason):

https://github.com/python/cpython/blob/7c0914d35eaaab2f323260ba5fe8884732533888/Lib/inspect.py#L1374-L1379

It's not a strong prefer for me, just a reminder, you could do a double check on this, if you have other idea on this, I'm ok with this.

Reply:
That make sense! thanks!